src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -980,15 +980,19 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 
 		try {
 
-			// Desktop DND (Import file)
+			// External file DND (Import/Upload file)
 			if (data instanceof NativeDragAndDropData) {
-				if (isWeb) {
+				// Native OS file DND into Web
+				if (originalEvent.dataTransfer?.types.indexOf('Files') !== -1 && isWeb) {
 					const browserUpload = this.instantiationService.createInstance(BrowserFileUpload);
 					await browserUpload.upload(target, originalEvent);
-				} else {
-					const nativeImport = this.instantiationService.createInstance(NativeFileImport);
-					await nativeImport.import(resolvedTarget, originalEvent);
 				}
+
+				// 2 Cases handled for import:
+				// FS-Provided file DND into Web/Desktop
+				// Native OS file DND into Desktop
+				const fileImport = this.instantiationService.createInstance(ExternalFileImport);
```

Comment:
Should this run even when we did run the `BrowserFileUpload`? Isn't it the one or the other?